### PR TITLE
BLUE-130 Global txreceipt forwarding

### PR DESCRIPTION
**Pull request by jairajdev**

### CODE REVIEW

File: src/state-manager/TransactionQueue.ts [hunk 1]
```diff
@@ -7497,23 +7499,43 @@ class TransactionQueue {
     // this.updateTxState(queueEntry, 'almostExpired')
     queueEntry.almostExpired = true
 
-    /* prettier-ignore */ nestedCountersInstance.countEvent( 'txAlmostExpired', `tx: ${this.app.getSimpleTxDebugValue(queueEntry.acceptedTx?.data)}` )
+    /* prettier-ignore */
+    nestedCountersInstance.countEvent("txAlmostExpired", `tx: ${this.app.getSimpleTxDebugValue(queueEntry.acceptedTx?.data)}`)
   }
 
-  getArchiverReceiptFromQueueEntry(queueEntry: QueueEntry): ArchiverReceipt | null {
+  async getArchiverReceiptFromQueueEntry(queueEntry: QueueEntry): Promise<ArchiverReceipt> {
```
Comment (arhamj) [inline src/state-manager/TransactionQueue.ts:7506]: This change is unrelated to the project. Please make sure this has been tested.

File: src/state-manager/TransactionQueue.ts [hunk 2]
```diff
@@ -7708,8 +7848,93 @@ class TransactionQueue {
     profilerInstance.profileSectionEnd('requestFinalData')
   }
 
+  async requestInitialData(queueEntry: QueueEntry, accountIds: string[]): Promise<WrappedResponses> {
```
Comment (arhamj) [inline src/state-manager/TransactionQueue.ts:7851]: This change is unrelated to the project. Please make sure this has been tested.

File: src/state-manager/TransactionQueue.ts [hunk 3]
```diff
@@ -7708,8 +7848,93 @@ class TransactionQueue {
...
+        this.mainLogger.error(`requestInitialData: txid: ${queueEntry.logID} error: ${e.message}`)
+      }
+    }
+    /* prettier-ignore */ this.mainLogger.error(`requestInitialData: txid: ${queueEntry.logID} failed. successCount: ${successCount} accountIds: ${accountIds.length}`);
+    profilerInstance.profileSectionEnd('requestInitialData')
+  }
+
   resetReceiptsToForward(): void {
-    const MAX_RECEIPT_AGE_MS = 25000 // 25s
+    const MAX_RECEIPT_AGE_MS = 15000 // 15s
```
Comment (arhamj) [inline src/state-manager/TransactionQueue.ts:7937]: Is this change intentional? Please confirm with @afostr

File: src/state-manager/index.ts [hunk 4]
```diff
@@ -1859,11 +1861,111 @@ class StateManager {
       },
     }
 
+    const requestTxAndStateBeforeBinaryHandler: Route<InternalBinaryHandler<Buffer>> = {
```
Comment (arhamj) [inline src/state-manager/index.ts:1864]: This change is unrelated to the project. Please make sure this has been tested.

**Comment by arhamj** [comment]
@S0naliThakur Please get more context on the commit `6019a8f` from Marc and Andrew and decide if we want to include these changes as part of shipping this project.

File: src/state-manager/TransactionQueue.ts (hunk 2 above)
Comment (S0naliThakur) [reply src/state-manager/TransactionQueue.ts:7851]: This was tested

File: src/state-manager/TransactionQueue.ts (hunk 1 above)
Comment (S0naliThakur) [reply src/state-manager/TransactionQueue.ts:7506]: tested

File: src/state-manager/index.ts (hunk 4 above)
Comment (S0naliThakur) [reply src/state-manager/index.ts:1864]: tested

**Review by arhamj — APPROVED** [review]

**Review by arhamj — DISMISSED** [review]

**Review by arhamj — APPROVED** [review]
